UofT-Frosh-Orientation/orientation-website
# 651 create schedule pdfpng

## **Contribution:**

### **Issue:**
Closes #651 

### **Accomplishments:**
- Displayed a button on the profile page to display PDF on click.
- Got a PDF to appear using `@react-pdf/renderer` which includes each day's events planned for Frosh and their location, date, and description.

### **Visuals:** 
Profile Page:
![image](https://github.com/UofT-Frosh-Orientation/orientation-website/assets/109243682/19ddb247-3109-4aaf-83c3-eb8430f687de)

PDF:
![image](https://github.com/UofT-Frosh-Orientation/orientation-website/assets/109243682/3927c07e-029f-4a80-b468-cff763e68922)


## **Details:**

### **Expected Behavior:**
- After logging in, heading to profile page, and clicking the download button, you should be able to view the Schedule PDF in a separate tab.

### **Testing Steps:**

Steps required to get expected behavior.
- Run the backend and log in using a frosh account or remove `<AuthorizedPage>` tags in `util/pages.jsx`
- Head to profile page
- Click on 'Download Schedule PDF' button
- Check if the PDF is displayed as expected

### **Complications:**
- Wanted to clarify whether to only show this button if `froshObject == true`? (its outside that check right now)
- Was trying to change the colour according to the 'Color' field in `src/assets/data.jsx`. For some reason, it wouldn't work when I tried to put it as the background colour, so I just left it as some default purple in the stylesheet for now.
- There's some unknown characters in the event dates (shows as '//' in the PDF) that I'm not sure how to get rid of.
- Couldn't figure out how to use custom fonts or make font bold/italicized. I attempted registering the font but got a whole bunch of errors.

### **Resources:**
- ReactPDF font: [https://snyk.io/advisor/npm-package/@react-pdf/renderer/functions/@react-pdf%2Frenderer.Font.register](url)
